src/libraries/System.Private.Xml/src/System/Xml/XmlConvert.cs

```diff
@@ -1645,5 +1645,174 @@ internal static Exception CreateInvalidNameCharException(string name, int index,
         {
             return CreateException(index == 0 ? SR.Xml_BadStartNameChar : SR.Xml_BadNameChar, XmlException.BuildCharExceptionArgs(name, index), exceptionType, 0, index + 1);
         }
+
+        internal static bool TryFormat(bool value, Span<char> destination, out int charsWritten)
+        {
+            ReadOnlySpan<char> valueSpan = value ? "true" : "false";
+
+            charsWritten = valueSpan.Length;
+            return valueSpan.TryCopyTo(destination);
+        }
+
+        internal static bool TryFormat(char value, Span<char> destination, out int charsWritten)
+        {
+            charsWritten = -1;
+            if (destination.Length < 1) return false;
```

Comment:
Do we need to set an invalid number for charsWritten when we're returning false? The pattern is that the out parameter is undefined when we return false, so why not always set it to 1 and save an assignment.

Reply:
done